src/Datum.ts

```diff
@@ -275,18 +280,81 @@ export function getOrd<A>(O: Ord<A>): Ord<Datum<A>> {
  * `Apply` semigroup
  *
  * @since 2.0.0
+ * 
+ * Note: Semantics changed in 4.0.0 to match new Apply instance
  */
-export const getApplySemigroup = <A>(S: Semigroup<A>): Semigroup<Datum<A>> =>
-  getSemigroup(S);
+export const getApplySemigroup = <A>(S: semigroup.Semigroup<A>): semigroup.Semigroup<Datum<A>> => ({
+  // TODO: replace with apply.getApplySemigroup if we bump the min supported version of fp-ts >= 2.10
+  // Or just remove in place of that factory instance
+  concat: (first: Datum<A>, second: Datum<A>) =>
+    Apply.ap(
+      Apply.map(first, (x: A) => (y: A) => S.concat(x, y)),
+      second
+    )
+})
 
 /**
  * @since 2.0.0
+ * 
+ * @deprecated Does not match getApplySemigroup. Use getMonoid for the same behavior under a new name.
  */
-export const getApplyMonoid = <A>(M: Monoid<A>): Monoid<Datum<A>> => ({
-  ...getApplySemigroup(M),
+export const getApplyMonoid = <A>(M: Monoid<A>): Monoid<Datum<A>> => getMonoid(M)
+
+/**
+ * @since 4.0.0
+ * 
+ */
+ export const getMonoid = <A>(M: Monoid<A>): Monoid<Datum<A>> => ({
+  ...getSemigroup(M),
   empty: replete(M.empty)
 });
 
+/**
+ * Viewing Datum as the following progess of state changes:
+ * 
+ * Initial -> Pending -> Replete -> Refresh [-> Replete -> ...]
+ * 
+ * This semigroup has a bias towards datums with values and a bias towards Pending/Refresh. 
+ * Notably, concat(Pending, Replete) gives Refresh.
+ * If both datums have a value, they're combined with the given Semigroup instance.
+ * 
+ * @since 4.0.0
+ */
+export const getProgressSemigroup = <A>(S: semigroup.Semigroup<A>): semigroup.Semigroup<Datum<A>> => ({
```

Comment:
This is entirely new, as proposed here:
https://github.com/nullpub/datum/issues/16

I think this is a generally useful instance... I also selfishly find myself needing it every time I use this library so I added it.

I can rip this out into it's own PR and/or change the name if you'd prefer.

Reply:
Looks good to me! See above naming changes.